### PR TITLE
Support more symbolic decompositions

### PR DESCRIPTION
**Pull request by anurudhp**
- permutation bloqs: decompose for symbolic `N`
- stateprep rotation - check phase bitsize, raise `DecomposeTypeError` when not possible.

### CODE REVIEW

**Review by tanujkhattar — APPROVED** [review]

**Comment by tanujkhattar** [comment]
I think it'll be nice to add some tests, but LGTM overall.

**Comment by mpharrigan** [comment]
yeah, this could use some symbolic `bloq_example`s

**Comment by anurudhp** [comment]
Added some partially symbolic bloq examples along with a `assert_valid_decomposition` test for them

File: qualtran/bloqs/arithmetic/permutation_test.py [hunk 1]
```diff
@@ -44,16 +48,38 @@
...
+        _permutation,
+        _sparse_permutation,
+        _permutation_cycle_symb_N,
+        _permutation_symb_with_cycles,
+        _sparse_permutation_with_symbolic_N,
+    ],
+    ids=lambda bloq_ex: bloq_ex.name,
+)
+def test_decomposition(bloq_ex):
+    qlt_testing.assert_valid_bloq_decomposition(bloq_ex.make())
```
Comment (mpharrigan) [inline qualtran/bloqs/arithmetic/permutation_test.py:82]: this is redundant with the `bloq_autotester`

Comment (anurudhp) [reply qualtran/bloqs/arithmetic/permutation_test.py:82]: I want to verify that the bloq actually decomposes (i.e. without raising a `DecomposeTypeError`). `bloq_autotester` treats `DecomposeTypeError` as a success right?

Comment (mpharrigan) [reply qualtran/bloqs/arithmetic/permutation_test.py:82]: oh good point. (technically it would report it as a skip)

**Pull request merged**